src/renderer/screens/exchange/Swap2/History/History.js

```diff
@@ -0,0 +1,181 @@
+  const dispatch = useDispatch();
+
+  const onExportOperations = useCallback(() => {
+    async function asyncExport() {
+      const path = await remote.dialog.showSaveDialog({
+        title: "Exported swap history",
+        defaultPath: `ledgerlive-swap-history-${moment().format("YYYY.MM.DD")}.csv`,
+        filters: [
+          {
+            name: "All Files",
+            extensions: ["csv"],
+          },
+        ],
+      });
+
+      if (path && mappedSwapOperations) {
+        exportOperations(path, mappedSwapOperationsToCSV(mappedSwapOperations), () =>
+          setExporting(false),
+        );
+      }
+    }
+    if (!exporting) {
+      asyncExport()
+        .catch(e => {
+          console.log({ e });
+        })
+        .then(() => {
+          setExporting(false);
+        });
+    }
+  }, [exporting, mappedSwapOperations]);
+
+  useEffect(() => {
+    (async function asyncGetCompleteSwapHistory() {
+      if (!accounts) return;
+      const sections = await getCompleteSwapHistory(flattenAccounts(accounts));
+      setMappedSwapOperations(sections);
+    })();
+  }, [accounts]);
+
+  const updateSwapStatus = useCallback(() => {
+    let cancelled = false;
+    async function fetchUpdatedSwapStatus() {
+      const updatedAccounts = await Promise.all(accounts.map(updateAccountSwapStatus));
+      if (!cancelled) {
+        updatedAccounts.filter(Boolean).forEach(account => {
+          dispatch(updateAccountWithUpdater(account.id, a => account));
+        });
+      }
+    }
+
+    fetchUpdatedSwapStatus();
+    return () => (cancelled = true);
+  }, [accounts, dispatch]);
+
+  const hasPendingSwapOperations = useMemo(() => {
+    if (mappedSwapOperations) {
+      for (const section of mappedSwapOperations) {
+        for (const swapOperation of section.data) {
+          if (operationStatusList.pending.includes(swapOperation.status)) {
+            return true;
+          }
+        }
+      }
+    }
+    return false;
+  }, [mappedSwapOperations]);
+
+  useInterval(() => {
+    if (hasPendingSwapOperations) {
+      updateSwapStatus();
+    }
+  }, 10000);
+
+  const openSwapOperationDetailsModal = useCallback(
+    mappedSwapOperation => setDrawer(SwapOperationDetails, { mappedSwapOperation }),
+    [],
+  );
+
+  return (
+    <Box p={20}>
+      <Box horizontal flow={2} alignItems="center" justifyContent="flex-end">
+        <ExportOperationsWrapper horizontal>
+          <IconDownloadCloud size={16} />
+          <Text ml={1} ff="Inter|Regular" fontSize={3}>
+            <FakeLink onClick={exporting ? undefined : onExportOperations}>
+              {exporting ? "Exporting..." : "Export operations"}
+            </FakeLink>
+          </Text>
+        </ExportOperationsWrapper>
+      </Box>
+      {mappedSwapOperations ? (
+        mappedSwapOperations.length ? (
+          <Box>
+            <Head px={20} py={16}>
+              <Alert type="primary">
+                <Trans i18nKey="swap.history.disclaimer" />
```

Comment:
swap2 all the history wording should be duplicated so we can just drop the old one when we finally switch

Reply:
Done! @juan-cortes